### PR TITLE
Hide Tokio crates behind cfg gate for Redox

### LINKED ISSUE #314: Fix Redox Build
tokio-signal brings in a large number of dependencies, many that do not support Redox yet. Using a lighter weight signal handling crate, or not linking to tokio-signal when compiling for Redox would help.

### PR DESCRIPTION
**Pull request by huntergoldstein**
Closes #314.

I'm still trying to set up a redox environment (see: https://github.com/redox-os/redox/issues/970; I'm giving up trying to do this locally for now, spinning up Ubuntu VM) but in the meantime this should work.

### CODE REVIEW

**Comment by jackpot51** [comment]
This looks good to me. Can you also change `Cargo.toml` ?

File: Cargo.toml [hunk 1]
```diff
@@ -27,6 +27,8 @@ permutate = "0.3"
 unicode-segmentation = "1.2"
 smallvec = "0.4"
 smallstring = "0.1"
+
+[target.'cfg(not(target_os = "redox"))'.dependencies]
```
Comment (jackpot51) [inline Cargo.toml]: Should this also contain `futures` ?

Comment (huntergoldstein) [inline Cargo.toml]: It should and it does now.

**Comment by mmstick** [comment]
Thanks! This should work for handling Redox.

**Pull request merged**